girder-dandi-archive/girder_dandi_archive/rest.py

```diff
@@ -69,21 +73,84 @@ def create_dandiset(self, params):
     @describeRoute(
         Description("Get Dandiset").param("identifier", "Dandiset Identifier", paramType="path")
     )
+    @dandiset_identifier
     def get_dandiset(self, identifier, params):
-
-        if not identifier:
-            raise RestException("identifier must not be empty.")
-
-        if not validate_dandiset_identifier(identifier):
-            raise RestException("Invalid Dandiset Identifier")
-
         # Ensure we are only looking for drafts collection child folders.
         drafts = get_or_create_drafts_collection()
         doc = Folder().findOne({"parentId": drafts["_id"], "meta.dandiset.identifier": identifier})
         if not doc:
             raise RestException("No such dandiset found.")
         return doc
 
+    @access.public
+    @describeRoute(
+        Description("Get Dandiset Owners").param(
+            "identifier", "Dandiset Identifier", paramType="path"
+        )
+    )
+    @dandiset_identifier
+    def get_dandiset_owners(self, identifier, params):
+        return Folder().getFullAccessList(find_dandiset_by_identifier(identifier))["users"]
+
+    @access.user
+    @autoDescribeRoute(
+        Description("Add Dandiset Owners")
+        .param("identifier", "Dandiset Identifier", paramType="path")
+        .jsonParam(
+            "owners",
+            "A JSON list of girder users to add as owners.",
+            paramType="body",
+            requireArray=True,
+        )
+    )
+    @dandiset_identifier
+    def add_dandiset_owners(self, identifier, owners, params):
+        dandiset = find_dandiset_by_identifier(identifier)
+        Folder().requireAccess(dandiset, user=self.getCurrentUser(), level=AccessType.ADMIN)
+
+        # Make sure the list doesn't contain duplicates
+        current_owners = [
```

Comment:
you aren't checking access levels in current owners, here or in other endpoints. could that have undesired effects?

it definitely means that `final_owners` could be mislabeled as it might include non-owners, which would be confusing.

Reply:
Since the call to `setAccessList` will replace entirely what's currently in the ACL, we have to include all users in the ACL in `final_owners`, in order to preserve any existing users, regardless of their access level.

Reply:
However I agree, the names are a bit misleading in that regard, so I can change them.